sdk/communication/Azure.Communication.Common/tests/CommunicationIdentifierTest.cs

```diff
@@ -111,5 +111,31 @@ public void RawIdIsOverriddenBySubTypes()
                 Assert.AreNotEqual(baseType, implementation.GetProperty(nameof(CommunicationIdentifier.RawId))?.DeclaringType);
             }
         }
+
+        [Test]
+        public void EqualityOperatorOverrideDoesntThrow()
+        {
+            var identifier = new CommunicationUserIdentifier("123");
```

Comment:
not sure if this is the case but we can try with a scenario where identifier is null (identifier = null). This should be passed as well.